### PR TITLE
4

### PR DESCRIPTION
**Pull request by Daniel-J-94**

### CODE REVIEW

File: controllers/getReviewByIdController.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const { getterReviewById } = require("../models/getReviewByIdModel.js");
+
+exports.getReviewById = (req, res, next) => {
+  const revIdFromReq = req.params;
+  getterReviewById(revIdFromReq)
+    .then((result) => {
+      console.log(
```
Comment (aloush-dev) [inline controllers/getReviewByIdController.js]: console log!

File: app.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const { getReviewById } = require("./controllers/getReviewByIdController");
+const app = express();
+
+app.get("/api/reviews/:review_id", getReviewById);
+app.use("*", (req, res) => {
+  res.status(404).send({ message: "Doesn't exist" });
+});
+app.use((err, req, res, next) => {
+  console.log(err);
+  res.status(404).send({ message: "No review with that ID" });
```
Comment (aloush-dev) [inline app.js]: The last error handler in your chain should be your 500 server error handler.
Currently any error that doesn't hit your previous error handlers comes here.
If there is no review with that ID what with results.rows look like and what do you need to do if it doesn't look how you're expecting

**Review by aloush-dev — COMMENTED** [review]
Your error handling isn't correct please fix before moving onto the next ticket.

**Review by aloush-dev — APPROVED** [review]

**Comment by Harrison365** [comment]
Hi Daniel :) 

This is getting there but a few more things are needed:
- Your /two request should return a 400, not 404, error status code. 
- You need to handle both errors with middleware functions PSQL error handlers and custom error handlers. 
- Your SQL string should not be a string literal. Just needs to be a regular string with $1 for the variable part. 

Please let us know if any of this needs clarifying. Give us a shout when you'd like us to check again 👍 
Thanks.

File: models/getReviewByIdModel.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const db = require("../db/connection");
+
+exports.fetchReviewById = (revIdFromReq) => {
+  const revIdNumber = revIdFromReq.review_id;
+
+  return db
+    .query("SELECT * FROM reviews WHERE review_id = $1;", [revIdNumber])
+
+    .then((result) => {
+      console.log("model result:", result);
```
Comment (itskatherine) [inline models/getReviewByIdModel.js]: console log begone

File: models/getReviewByIdModel.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const db = require("../db/connection");
+
+exports.fetchReviewById = (revIdFromReq) => {
```
Comment (itskatherine) [inline models/getReviewByIdModel.js]: nice clear naming conventions

File: controllers/getReviewByIdController.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const { fetchReviewById } = require("../models/getReviewByIdModel.js");
+
+exports.getReviewById = (req, res, next) => {
+  const revIdFromReq = req.params;
+  fetchReviewById(revIdFromReq)
+    .then((result) => {
+      console.log("result!", result);
```
Comment (itskatherine) [inline controllers/getReviewByIdController.js]: console log begone

File: controllers/getReviewByIdController.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const { fetchReviewById } = require("../models/getReviewByIdModel.js");
+
+exports.getReviewById = (req, res, next) => {
+  const revIdFromReq = req.params;
```
Comment (itskatherine) [inline controllers/getReviewByIdController.js]: You could take the review_id off the req.params here, rather than in the model

**Review by itskatherine — COMMENTED** [review]
Hi Daniel, a few minor changes but once you've addressed my comments you're good to merge 🙂 👍

**Pull request merged**